### PR TITLE
add aria label to spinner buttons

### LINKED ISSUE #2941: [BUG] Spinner buttons have a11y issue "no discernible text" - All Frameworks
## Describe The Bug
Storybook's a11y plugin reports an issue with our buttons with spinners in them. We should find a way to fix this.

## To Reproduce - ALL PACKAGES
1. Go to buttons in all packages storybook
2. Click on loading story
3. Scroll down to the tabs section and see the "Accessibility" panel results

## Expected Behavior - ALL PACKAGES
- Update JS, Add Aria Label 
- There would be no a11y error.

## Screenshots
Screen Shot 2020-03-16 at 11 14 20 AM

## Desktop (please complete the following information):
 - OS: all
 - Browser [e.g. chrome, safari]: all
 - Version [e.g. 22]: all

## Smartphone (please complete the following information):
 - Device: [e.g. iPhone6] all
 - OS: [e.g. iOS8.1] all
 - Browser [e.g. stock browser, safari] all
 - Version [e.g. 22] all

## Additional Context
Found this while adding a new story for secondary button spinners.

### PR DESCRIPTION
**Pull request by afebbraro**
## What does this PR do?
Add aria label to spinner buttons in all three frameworks and tests. Updates staging with master.

### Associated Issue
Fixes  #2941 

## Please check off completed items as you work.
If a checklist item or section does not apply to your PR
then please remove it.

### Documentation
 - [x] Update Spark Docs

### Code
 - [x] Build Component in HTML
 - [x] Build Component in Angular
 - [x] Build Component in React
 - [x] Unit Testing in HTML with `npm run test` in `html/` (100% coverage, 100% passing)
 - [x] Unit Testing in Angular with `npm run test` in `angular/` (100% coverage, 100% passing)
 - [x] Unit Testing in React with `npm run test` in `react/` (100% coverage, 100% passing)

### Accessibility
- [x] New changes abide by [accessibility requirements](https://sparkdesignsystem.com/docs/accessibility)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [ ] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [ ] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [ ] Apple Safari (Mobile)

### CODE REVIEW

File: html/components/button.stories.js [hunk 1]
```diff
@@ -25,6 +25,10 @@ always use a \`<button>\` element.
 must include a \`title=””\` attribute.
 - If a Button only includes an Icon with no text,
 alternative text must be provided.
+- If the Button has a spinner, the \`aria-label\`
+will be set to 'Loading' unless you supply a custom
+string value to the \`data-sprk-spinner-aria-label\`
+attribute.
 
 ##### Guidelines
 - If a Button is in a form, but is not intended to
```
Comment (bobwise) [inline html/components/button.stories.js]: The HTML story does not have any unit tests listed.

File: html/components/button.stories.js [hunk 2]
```diff
@@ -25,6 +25,10 @@ always use a \`<button>\` element.
 must include a \`title=””\` attribute.
 - If a Button only includes an Icon with no text,
 alternative text must be provided.
+- If the Button has a spinner, the \`aria-label\`
+will be set to 'Loading' unless you supply a custom
+string value to the \`data-sprk-spinner-aria-label\`
+attribute.
```
Comment (bobwise) [inline html/components/button.stories.js]: I would add something here that makes it clear this change only happens in the click event on the button, not when it is initialized.

File: react/src/components/buttons/SprkButton.js [hunk 3]
```diff
@@ -36,6 +37,7 @@ const SprkButton = ({
       data-id={idString}
       data-analytics={analyticsString}
       disabled={disabled}
+      aria-label={loading ? spinningAriaLabel : undefined}
```
Comment (bobwise) [inline react/src/components/buttons/SprkButton.js]: This line is wiping out the initial value of aria-label when the component renders. It should only replace the value when loading is true.

File: html/components/button.stories.js (hunk 1 above)
Comment (afebbraro) [reply html/components/button.stories.js]: added! great catch

File: html/components/button.stories.js (hunk 2 above)
Comment (afebbraro) [reply html/components/button.stories.js]: ohh good idea!

File: react/src/components/buttons/SprkButton.js (hunk 3 above)
Comment (afebbraro) [reply react/src/components/buttons/SprkButton.js]: great catch, updated!

**Pull request merged**